### ISSUE TITLE
Citation Parser Not Grabbing all Citations

### ISSUE THREAD

**Issue by Colinstarger** [freelawproject/courtlistener#540]
Right now, the citation parser is not picking up some seemingly simple cites. A great example is at the very bottom of this case (Lochner): https://www.courtlistener.com/opinion/102784/west-coast-hotel-co-v-parrish/

To give just one example, consider this line in footnote 1.

[1] Allgeyer v. Louisiana, 165 U.S. 578; Lochner v. New York, 198 U.S. 45; Adair v. United States, 208 U.S. 161.

The parser DOES NOT pick up Allgeyer or Lochner, but it does pick up Adair. Footnote 2 shows a similar problem.

**Comment by Colinstarger** [comment]
Just giving another example BROWN v. BOARD (1954). https://www.courtlistener.com/opinion/105221/brown-v-board-of-education/

Has this paragraph:
"In the first cases in this Court construing the Fourteenth Amendment, decided shortly after its adoption, the Court interpreted it as proscribing all state-imposed discriminations against the Negro race.[5] The doctrine of *491 "separate but equal" did not make its appearance in this Court until 1896 in the case of Plessy v. Ferguson, supra, involving not education but transportation.[6] American courts have since labored with the doctrine for over half a century. In this Court, there have been six cases involving the "separate but equal" doctrine in the field of public education.[7] In Cumming v. County Board of Education, 175 U.S. 528, and Gong Lum v. Rice, 275 U.S. 78, the validity of the doctrine itself was not challenged.[8] In more recent cases, all on the graduate school *492 level, inequality was found in that specific benefits enjoyed by white students were denied to Negro students of the same educational qualifications. Missouri ex rel. Gaines v. Canada, 305 U. S. 337; Sipuel v. Oklahoma, 332 U. S. 631; Sweatt v. Painter, 339 U. S. 629; McLaurin v. Oklahoma State Regents, 339 U. S. 637. In none of these cases was it necessary to re-examine the doctrine to grant relief to the Negro plaintiff. And in Sweatt v. Painter, supra, the Court expressly reserved decision on the question whether Plessy v. Ferguson should be held inapplicable to public education."
  
Alas, All the cases after "educational qualifications" are not picked up. I am going to add them by hand now for another project. But the error is befuddling

**Comment by mlissner** [comment]
This looks like punctuation and semi-colons are causing it. Might be an easy fix.

**Comment by mlissner** [comment]
Starting in on the issues in the following text: 

>  Munn v. Illinois, 94 U.S. 113; Railroad Commission Cases, 116 U.S. 307; Willcox v. Consolidated Gas Co., 212 U.S. 19; Atkin v. Kansas, 191 U.S. 207; Mugler v. Kansas, 123 U.S. 623; Crowley v. Christensen, 137 U.S. 86; Gundling v. Chicago, 177 U.S. 183; Booth v. Illinois, 184 U.S. 425; Schmidinger v. Chicago, 226 U.S. 578; Armour & Co. v. North Dakota, 240 U.S. 510; National Fire Insurance Co. v. Wanberg, 260 U.S. 71; Radice v. New York, 264 U.S. 292; Yeiser v. Dysart, 267 U.S. 540; Liberty Warehouse Co. v. Burley Tobacco Growers' Assn., 276 U.S. 71, 97; Highland v. Russell Car Co., 279 U.S. 253, 261; O'Gorman & Young v. Hartford Insurance Co., 282 U.S. 249, 251; Hardware Dealers Insurance Co. v. Glidden Co., 284 U.S. 151, 157; Packer Corp. v. Utah, 285 U.S. 95, 111; Stephenson v. Binford, 287 U.S. 251, 274; Hartford Accident Co. v. Nelson Mfg. Co., 291 U.S. 352, 360; Petersen Baking Co. v. Bryan, 290 U.S. 570; Nebbia v. New York, 291 U.S. 502, 527-529.

First test is to run the citation parser against it. It runs successfully finding everything. Next step: See if the matching algo works for the citations that were found.

**Comment by mlissner** [comment]
OK, digging in more, I couldn't find any issues, so I just re-ran the citation parser, and sure enough, every single citation in that footnote was found. Looks like this issue has been fixed already, but that the citation parser hadn't yet been run for it.

Moving onto *Brown*, I did the same kinds of tests, and then eventually just re-ran the parser. Sure enough, it found all the cases that were previously missing. 

This is unsatisfying, but I think this is fixed already, just not reflected in the current parse of the corpus.
